### PR TITLE
Merge {IO,Environment,Windows}Error into OSError for Python 3.3+

### PR DESCRIPTION
**Pull request by intgr**
Starting with Python 3.3, IOError, EnvironmentError and WindowsError are
aliases for OSError, which has all their attributes.

Also, tested on Python 3.6 that OSError.filename/filename2 may also be
bytes and None, so annotated them as Any.

Reference:
* https://docs.python.org/3/library/exceptions.html#OSError
* https://www.python.org/dev/peps/pep-3151/

### CODE REVIEW

File: stdlib/3/builtins.pyi [hunk 1]
```diff
@@ -937,11 +937,28 @@ class SystemExit(BaseException):
...
-    filename = ...  # type: str
+if sys.version_info >= (3, 3):
+    class OSError(Exception):
+        errno = 0
+        strerror = ...  # type: str
+        filename = ...  # type: Union[str, bytes, None]
+        if sys.version_info >= (3, 4):
+            filename2 = ...  # type: Union[str, bytes, None]
+        if sys.platform == 'win32':
+            winerror = 0
```
Comment (intgr) [inline stdlib/3/builtins.pyi]: Is it a good idea to put `sys.platform` checks in? The `winerror` attribute is missing entirely on non-Windows platforms.

File: stdlib/3/builtins.pyi [hunk 2]
```diff
@@ -937,11 +937,25 @@ class SystemExit(BaseException):
...
+    EnvironmentError = IOError = WindowsError = OSError
```
Comment (intgr) [inline stdlib/3/builtins.pyi]: D'oh, turns out that mypy doesn't allow class aliases... Should I make them subclasses even though they're not?

**Comment by srittau** [comment]
Question for the maintainers: Can't we leave out the branch for Python < 3.3? README.md says that typeshed targets Python 2.7 and 3.3+.

**Comment by intgr** [comment]
Thanks! I got rid of the Python <3.3 check.

It still fails a test in mypy code, which uses `OSError.filename`. The filename is being assigned to a variable that was inferred to `str` but fails since the attribute is now labelled `Union[str, bytes, None]`. See https://github.com/python/mypy/blob/master/mypy/test/data.py#L318-L323

What's the correct fix in this case? Should I submit a PR to mypy to create a new variable in this context? Or should I revert the type annotation to `str`? Or are there some other options?

**Comment by intgr** [comment]
I created a PR for Mypy here: https://github.com/python/mypy/pull/4541. I'll see whether that gets merged.

**Comment by intgr** [comment]
Per comment from GvR, I changed the `filename`/`filename2` type to Any instead of Union to avoid this issue. All tests pass now.

**Comment by ambv** [comment]
Thanks! ✨ 🍰 ✨

**Pull request merged**